src/Components/test/E2ETest/ServerExecutionTests/ProtectedBrowserStorageUsageTest.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            // Local storage initially cleared
+            var incrementLocalButton = Browser.FindElement(By.Id("increment-local"));
+            var localCount = Browser.FindElement(By.Id("local-count"));
+            Browser.Equal("0", () => localCount.Text);
+
+            // Local storage updates in current tab
+            incrementLocalButton.Click();
+            Browser.Equal("1", () => localCount.Text);
+
+            // Local storage persists accross tabs
+            OpenNewSession();
+            localCount = Browser.FindElement(By.Id("local-count"));
+            Browser.Equal("1", () => localCount.Text);
+        }
+
+        [Fact]
+        public void SessionStoragePersistsOnRefresh()
+        {
+            // Session storage initially cleared
+            var incrementSessionButton = Browser.FindElement(By.Id("increment-session"));
+            var sessionCount = Browser.FindElement(By.Id("session-count"));
+            Browser.Equal("0", () => sessionCount.Text);
+
+            // Session storage updates
+            incrementSessionButton.Click();
+            Browser.Equal("1", () => sessionCount.Text);
+
+            // Session storage persists on refresh
+            Browser.Navigate().Refresh();
+            Browser.MountTestComponent<ProtectedBrowserStorageUsageComponent>();
+
+            sessionCount = Browser.FindElement(By.Id("session-count"));
+            Browser.Equal("1", () => sessionCount.Text);
+        }
+
+        [Fact]
+        public void SessionStorageDoesNotPersistAccrossTabs()
+        {
+            // Session storage initially cleared
+            var incrementSessionButton = Browser.FindElement(By.Id("increment-session"));
+            var sessionCount = Browser.FindElement(By.Id("session-count"));
+            Browser.Equal("0", () => sessionCount.Text);
+
+            // Session storage updates in current tab
+            incrementSessionButton.Click();
+            Browser.Equal("1", () => sessionCount.Text);
+
+            // Session storage does not persist accross tabs
+            OpenNewSession();
+            sessionCount = Browser.FindElement(By.Id("session-count"));
+            Browser.Equal("0", () => sessionCount.Text);
+        }
+
+        /// <summary>
+        /// Opens a new session in a new tab, mounting a new test component.
+        /// </summary>
+        /// <remarks>
+        /// Simply opening a new tab using JS is not sufficient because the browser context perists and
+        /// the same session is maintained. The way this method starts a new session is by simulating a
+        /// ctrl+click (or command+click on Mac) on a link that opens a new tab. This opens it as a background
+        /// tab, which has a new browser context.
+        /// </remarks>
+        private void OpenNewSession()
+        {
+            var modifierKey = Browser is RemoteWebDriver remoteBrowser && remoteBrowser.Capabilities.Platform.PlatformType == PlatformType.Mac ?
```

Comment:
Would it suffice to check if the current OS is mac? `if (RuntimeInformation.IsOSPlatform)?`

Reply:
Hmm, I don't think we need this since we only run these tests on windows, but I'm fine leaving it here if in the future we change our minds.